channeld/channeld.c

```diff
@@ -2199,40 +2586,1304 @@ static void handle_unexpected_reestablish(struct peer *peer, const u8 *msg)
+				      &peer->channel->funding_pubkey[LOCAL],
+				      &splice_sig))
+		status_failed(STATUS_FAIL_INTERNAL_ERROR,
+			      "Unable to set signature internally "
+			      "funding_index: %d "
+			      "my pubkey: %s "
+			      "my signature: %s "
+			      "psbt: %s",
+			      splice_funding_index,
+			      type_to_string(tmpctx, struct pubkey, &peer->channel->funding_pubkey[LOCAL]),
+			      type_to_string(tmpctx, struct bitcoin_signature, &splice_sig),
+			      type_to_string(tmpctx, struct wally_psbt, signed_psbt));
+
+	txsig_tlvs = tlv_txsigs_tlvs_new(tmpctx);
+	der_len = signature_to_der(der, &splice_sig);
+	txsig_tlvs->funding_outpoint_sig = tal_dup_arr(tmpctx, u8, der,
+						       der_len, 0);
+
+	ws = psbt_to_witness_stacks(tmpctx, signed_psbt,
+				    TX_INITIATOR, splice_funding_index);
+	sigout_msg = towire_tx_signatures(tmpctx, &peer->channel_id,
+					  &signed_psbt_txid, ws, txsig_tlvs);
+
+	if (do_i_sign_first(peer, signed_psbt, TX_INITIATOR)) {
+		status_debug("Splice initiator: we sign first");
+		outmsg = towire_channeld_update_inflight(NULL, signed_psbt);
+		wire_sync_write(MASTER_FD, take(outmsg));
+		peer_write(peer->pps, take(sigout_msg));
+	}
+
+	msg = peer_read(tmpctx, peer->pps);
+	if (handle_peer_error(peer->pps, &peer->channel_id, msg))
+		return;
+
+	if (fromwire_peektype(msg) != WIRE_TX_SIGNATURES)
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				"Splicing got incorrect message from peer: %s "
+				"(should be WIRE_TX_SIGNATURES)",
+				peer_wire_name(fromwire_peektype(msg)));
+
+	if (!fromwire_tx_signatures(tmpctx, msg, &cid, &txid,
+				    cast_const3(struct witness_stack ***, &ws),
+				    &txsig_tlvs))
+		peer_failed_warn(peer->pps, &peer->channel_id,
+			    "Splicing bad tx_signatures %s",
+			    tal_hex(msg, msg));
+
+	their_sig.sighash_type = SIGHASH_ALL;
+
+	if (!signature_from_der(txsig_tlvs->funding_outpoint_sig,
+			       tal_count(txsig_tlvs->funding_outpoint_sig),
+			       &their_sig)) {
+
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "Splicing bad tx_signatures %s",
+				 tal_hex(msg, msg));
+	}
+
+	their_pubkey = &peer->channel->funding_pubkey[REMOTE];
+
+	/* Set the commit_sig on the commitment tx psbt */
+	if (!psbt_input_set_signature(signed_psbt,
+				      splice_funding_index,
+				      their_pubkey,
+				      &their_sig)) {
+
+		status_failed(STATUS_FAIL_INTERNAL_ERROR,
+			      "Unable to set signature internally "
+			      "funding_index: %d "
+			      "pubkey: %s "
+			      "signature: %s "
+			      "psbt: %s",
+			      splice_funding_index,
+			      type_to_string(tmpctx, struct pubkey, their_pubkey),
+			      type_to_string(tmpctx, struct bitcoin_signature, &their_sig),
+			      type_to_string(tmpctx, struct wally_psbt, signed_psbt));
+	}
+
+	psbt_input_set_witscript(signed_psbt,
+				 splice_funding_index,
+				 wit_script);
+
+	final_tx = bitcoin_tx_with_psbt(tmpctx, signed_psbt);
```

Comment:
should we check that the psbt is finalized before converting to tx?

Reply:
The `signed_psbt` isn't finalized yet because it's still missing one signature. We're generating the bitcoin_tx so that we can pass it to libhsmd and get the signature, to then make it final.